### ISSUE TITLE
Implement `sits_smooth()` to probability vector cubes (`probs_vector_cube`)

### ISSUE THREAD

**Issue by rolfsimoes** [e-sensing/sits#1506]
**Describe the requested improvement**

Add support for segment-based Bayesian smoothing in `sits_smooth()`. Currently, the function only processes raster probability cubes (`probs_cube`) using a spatial rolling window. 

Object-Based Image Analysis (OBIA) workflows require smoothing at the segment level. 
The new method must use segment geometries (polygons) from a `probs_vector_cube` to define pixel neighborhoods.

**Associated sits API function**

`sits_smooth.probs_vector_cube()`

Signature:

```R
sits_smooth.probs_vector_cube <- function(cube, ...,
                                          neigh_fraction = 0.5,
                                          smoothness = 20.0,
                                          memsize = 4L,
                                          multicores = 2L,
                                          output_dir,
                                          version = "v1",
                                          progress = TRUE)
```

The function should:
1. Extract pixel probabilities inside each segment geometry.
2. Convert probabilities to logits.
3. Apply Bayesian smoothing per segment.
4. Convert logits back to probabilities.
5. Save the smoothed probability raster and return a `probs_vector_cube`.

**Additional context**

This implementation aligns `sits_smooth()` with other vector cube operations such as `sits_variance()` and `sits_uncertainty()`. The output cube can be used directly in `sits_label_classification()`.
